corehq/apps/es/management/commands/es_version_for_index.py

```diff
@@ -0,0 +1,45 @@
+from django.core.management.base import BaseCommand
+from corehq.apps.es.client import ElasticManageAdapter
+from corehq.util.markup import SimpleTableWriter, TableRowFormatter
+
+
+class Command(BaseCommand):
+    help = ("Prints the ES version with which each index was created")
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--only-older-version-indexes',
+            default=False,
+            action='store_true',
+            help="""Would only print indexes that were
+            created with versions older than currently running ES""",
+        )
+
+    def _normalize_version(self, version):
+        """
+        Transforms es version string to human readable form
+        """
+        return f"{version[0]}.{version[2]}.{version[4]}"
+
+    def handle(self, only_older_version_indexes, **options):
+        es_manager = ElasticManageAdapter()
+        indices = es_manager.get_indices()
+        current_es_version = es_manager.info()['version']['number']
+
+        rows = []
+        for index_name, index_info in indices.items():
+            version = self._normalize_version(index_info["settings"]["index"]["version"]["created"])
```

Comment:
It might be nice if we had a utility that got the version and returned it in a `namedtuple` or `attrs` class or similar so that the different components were named instead of being called by inscrutable index. This could probably help the adapter classes that do the same thing as well.

Reply:
That is a good suggestion but we already have a function in the adapter that returns the ES version in form of tuple https://github.com/dimagi/commcare-hq/blob/jm+ap/elastic-multiplexer/corehq/apps/es/client.py#L62

And I am not sure if we need an Index creation version for programmatic use as of now but if the requirement will come up would be going with your suggestion.